opentelemetry-api/src/opentelemetry/util/_importlib_metadata.py

```diff
@@ -12,27 +12,163 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# type: ignore
+
 from sys import version_info
+from typing import Dict, Tuple, Union
+
+# This is a cache to avoid going through creating a dictionary for all entry
+# points for 3.7 every time the entry points function is called.
+result_entry_points_37 = {}
 
 # FIXME remove this when support for 3.7 is dropped.
 if version_info.minor == 7:
     # pylint: disable=import-error
-    from importlib_metadata import entry_points, version  # type: ignore
+    from importlib_metadata import EntryPoint
+    from importlib_metadata import (
+        entry_points as importlib_metadata_entry_points,
+    )
+    from importlib_metadata import version
+
+    def entry_points(
```

Comment:
Shouldn't there be a single function definition `entry_points(...)` and then the logic branching for versions happen within the function?

Reply:
That would make a version check every time the function is called, this way versions are only checked once.

Reply:
@ocelotl 

That is true, please add explicit typing definitions for each of the function defs to make it more obvious what the functions return.

Reply:
Fixed